### PR TITLE
fix(data-grid): allow clicking links in the data-grid

### PR DESCRIPTION
**Pull request by TheSisb**
The onMouseDown was swallowing the onClick event on the child anchor tag (https://github.com/twilio-labs/paste/issues/2061). DataGridCell doesn't need to ensure focus onMouseDown, it can operate onClick, which fixes this issue.

### CODE REVIEW

**Comment by shleewhite** [comment]
@TheSisb should there be tests to make sure the child onclick is working correctly?

File: packages/paste-core/components/data-grid/stories/components/ComposableCellsDataGrid.tsx [hunk 1]
```diff
@@ -15,7 +15,9 @@ const ActionMenu: React.FC = () => {
         <MoreIcon decorative={false} title="More options" />
       </MenuButton>
       <Menu {...menu} aria-label="Preferences">
-        <MenuItem {...menu}>Settings</MenuItem>
+        <MenuItem {...menu} href="https://google.com">
```
Comment (gloriliale) [inline packages/paste-core/components/data-grid/stories/components/ComposableCellsDataGrid.tsx:18]: Good call for putting this in the story 🎉

**Review by zahnster — APPROVED** [review]
Looks good!

**Review by gloriliale — APPROVED** [review]
Excited that this fixes it! Other than +1 on @shleewhite's test question, looks good to me!

**Comment by TheSisb** [comment]
@shleewhite I could add one but didn't for a few reasons:
1. I wanted to get this out ASAP before the holiday PTO.
2. It's a very visible break that is highly unlikely to resurface without us noticing.

If you feel strongly about adding a test, I can try to squeeze it in.

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**